plugins/ingest/src/main/java/org/elasticsearch/ingest/processor/mutate/MutateProcessor.java

```diff
@@ -0,0 +1,312 @@
+        }
+    }
+
+    private void doRename(Data data) {
+        if (rename == null) { return; }
+
+        for(Map.Entry<String, String> entry : rename.entrySet()) {
+            Object oldVal = data.getProperty(entry.getKey());
+            data.addField(entry.getValue(), oldVal);
+        }
+    }
+
+    private Object parseValueAsType(Object oldVal, String toType) {
+        if("integer".equals(toType)) {
+            oldVal = Integer.parseInt(oldVal.toString());
+        } else if("float".equals(toType)) {
+            oldVal = Float.parseFloat(oldVal.toString());
+        } else if("string".equals(toType)) {
+            oldVal = oldVal.toString();
+        } else if("boolean".equals(toType)) {
+            String val = oldVal.toString();
+            if (val.equals("true") || val.equals("t") || val.equals("yes") || val.equals("y") || val.equals("1")) {
+                oldVal = true;
+            } else if (val.equals("false") || val.equals("f") || val.equals("no") || val.equals("n") || val.equals("0")) {
+                oldVal = false;
+            } else {
+                throw new IllegalArgumentException("Cannot convert value: " + oldVal.toString() + "into a boolean");
+            }
+        }
+        return oldVal;
+    }
+
+    @SuppressWarnings("unchecked")
+    private void doConvert(Data data) {
+        if (convert == null) { return; }
+
+        for(Map.Entry<String, String> entry : convert.entrySet()) {
+            String toType = entry.getValue();
+
+            Object oldVal = data.getProperty(entry.getKey());
+            Object newVal;
+
+            if (oldVal instanceof List) {
+                newVal = new ArrayList<>();
+                for (Object e : ((List<Object>) oldVal)) {
+                    ((List<Object>) newVal).add(parseValueAsType(e, toType));
+                }
+            } else {
+                newVal = parseValueAsType(oldVal, toType);
+            }
+
+            data.addField(entry.getKey(), newVal);
+        }
+    }
+
+    private void doSplit(Data data) {
+        if (split == null) { return; }
+
+        for(Map.Entry<String, String> entry : split.entrySet()) {
+            Object oldVal = data.getProperty(entry.getKey());
+            if (oldVal instanceof String) {
+                data.addField(entry.getKey(), Arrays.asList(((String) oldVal).split(entry.getValue())));
+            } else {
+                throw new IllegalArgumentException("Cannot split a field that is not a String type");
+            }
+        }
+    }
+
+    private void doGsub(Data data) {
+        if (gsub == null) { return; }
+
+        for (Map.Entry<String, List<String>> entry : gsub.entrySet()) {
+            String fieldName = entry.getKey();
+            List<String> matchAndReplaceList = entry.getValue();
+            String oldVal = data.getProperty(fieldName);
+            String newVal = oldVal.replaceAll(matchAndReplaceList.get(0), matchAndReplaceList.get(1));
```

Comment:
ah, yeah. I'll see what I can do to maybe clean up the gsub variable in the Processor to contain this Processor. will have to change some of that structure. or keep a separate list of Pattern objects. I'll update with the changes.

same would apply to other methods that deal with regex expressions like `split`